In [1]:
import pandas as pd
import numpy as np

In [2]:
data1 = pd.read_excel('/content/drive/MyDrive/DSA WORKS/Week 6/Dataset_w6.xlsx')

In [3]:
data1

country  ... International trade: Imports (million US$)
0                           Afghanistan  ...                                       3568
1                               Albania  ...                                       4669
2                               Algeria  ...                                      47091
3                                Angola  ...                                       8790
4                   Antigua and Barbuda  ...                                        491
..                                  ...  ...                                        ...
191  Venezuela (Bolivarian Republic of)  ...                                      16324
192                            Viet Nam  ...                                     174111
193                               Yemen  ...                                       6861
194                              Zambia  ...                                       7442
195                            Zimbabwe  ...                                       5212

[196 rows x 7 columns]

**1.Find out how many countries are there in the dataset.**


In [4]:
data1['country'].nunique()

196

There are **196** countries in the given dataset

**4.a On Average which region has the highest GDP per capita (current US%)**

In [38]:
gdpdata=data1.groupby(['Region'])[['GDP per capita (current US$)','International trade: Imports (million US$)']].mean()
gdpdata['GDP per capita (current US$)'].sort_values(ascending=False)

Region
NorthernAmerica      57659.000000
WesternEurope        55387.128571
Oceania              44823.250000
NorthernEurope       42098.220000
EasternAsia          27967.614286
WesternAsia          17013.938889
Caribbean            15135.562500
SouthernEurope       14462.741667
South-easternAsia    10299.027273
EasternEurope         9292.490000
Melanesia             9261.960000
Polynesia             8686.520000
SouthAmerica          8499.775000
CentralAmerica        6348.137500
Micronesia            5008.300000
CentralAsia           4329.880000
SouthernAfrica        4191.900000
NorthernAfrica        3934.960000
MiddleAfrica          3928.833333
SouthernAsia          2968.400000
EasternAfrica         2088.852941
WesternAfrica         1013.140000
Name: GDP per capita (current US$), dtype: float64

on evaluation of the data the region having highest GDP on region wise is ***NORTHERNAMERICA*** as shown above with ***57659*** US$ as avegrage value of GDP

**4.b On Average which region has the highest International trade: Imports (million US$)**

In [39]:
gdpdata['International trade: Imports (million US$)'].sort_values(ascending=False)

Region
NorthernAmerica      663552.250000
EasternAsia          451995.000000
WesternEurope        407498.428571
NorthernEurope       113407.200000
Oceania              112914.500000
South-easternAsia     98324.636364
EasternEurope         85243.900000
SouthernEurope        76666.666667
SouthernAsia          58199.333333
CentralAmerica        57031.500000
WesternAsia           46025.555556
NorthernAfrica        35034.400000
SouthAmerica          32885.666667
SouthernAfrica        18005.600000
CentralAsia           12063.400000
WesternAfrica          4898.333333
EasternAfrica          4726.176471
MiddleAfrica           4381.222222
Caribbean              3057.312500
Melanesia              2896.400000
Polynesia               432.000000
Micronesia               99.000000
Name: International trade: Imports (million US$), dtype: float64

On evaluation of the data the region having highest International trade: Imports on region wise is **NORTHERNAMERICA**as shown above with **663552.25** millon US$ as average